var/spack/repos/builtin/packages/op2-dsl/package.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Op2Dsl(MakefilePackage):
+    """OP2 is a high-level embedded domain specific language for writing
+    unstructured mesh algorithms with automatic parallelisation on multi-core
+    and many-core architectures."""
+
+    homepage = "https://op-dsl.github.io/"
+    git      = "https://github.com/OP-DSL/OP2-Common.git"
+
+    maintainers = ['gihanmudalige', 'reguly', 'bozbez']
+
+    version('master', branch='master')
+    version('1.1.0', tag='v1.1.0')
+
+    build_directory = 'op2'
+
+    variant('mpi', default=True, description='Enable MPI support')
+    variant('cuda', default=False, description='Enable CUDA support')
+
+    variant('parmetis', default=True,
+            description='Enable ParMETIS partitioning support for MPI')
+
+    variant('scotch', default=True,
+            description='Enable PT-Scotch partitioning support for MPI')
+
+    depends_on('cuda', when='+cuda')
```

Comment:
Sorry, didn't notice until now: do you have a particular reason not to extend CudaPackage? It defines the cuda variant too, plus a bunch of compiler conflicts.

Reply:
Good point, I'll need to push some changes to the source repo first though to make `cuda_arch` work out properly first.

Reply:
Right, it's only relevant if you actually compile device code, if you just link to cuda libs, it's fine to leave it like this.

Reply:
We do compile device code, the problem is the way we specify the build architecture is a bit awkward (`NV_ARCH={Kepler, Maxwell, ...}`) - I was thinking about getting around to improving this anyway, so might as well do it now.